src/cli/cli.cpp

```diff
@@ -136,14 +155,51 @@ const struct Command Interpreter::sCommands[] =
     { "whitelist", &Interpreter::ProcessWhitelist },
 };
 
+#ifdef OTDLL
+uint32_t otPlatRandomGet(void)
+{
+    return (uint32_t)rand();
+}
+#else
+void otFreeMemory(const void *)
+{
+    // No-op on systems running OpenThread in-proc
+}
+#endif
+
+template <class T> class otPtr
```

Comment:
Can we have the class defined under OTDLL/Widnows only? And have typedefs like this:
```
#if OTDLL
typedef same as before
#else
typedef const otMacCounter *otMacCountersPtr;
....
#endif
```